src/components/TextField/Example.tsx

```diff
@@ -0,0 +1,107 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import TextField from './TextField';
+import { FieldValues, useForm } from 'react-hook-form';
+
+export default function TextFieldExample() {
+  const { handleSubmit, control, register } = useForm();
+  const [async, setStatus] = useState<{
+    status: 'default' | 'active' | 'success' | 'error' | 'loading';
+    message: string;
+  }>({ status: 'default', message: '' });
+  const ref = useRef<HTMLInputElement>(null);
+  const ref2 = useRef<HTMLInputElement>(null);
+  const ref3 = useRef<HTMLInputElement>(null);
+  const ref4 = useRef<HTMLInputElement>(null);
+
+  const onSubmit = (data: FieldValues) => {
+    console.log(data);
+  };
+
+  useEffect(() => {
+    setStatus({ status: 'loading', message: '로딩중입니다' });
+    setTimeout(() => {
+      setStatus({ status: 'error', message: '에러입니다' });
+    }, 1000);
+    setTimeout(() => {
+      setStatus({ status: 'success', message: '성공!' });
+    }, 2000);
+    setTimeout(() => {
+      setStatus({ status: 'error', message: '실패!' });
+    }, 3000);
+  }, []);
+
+  return (
+    <div>
+      <div
+        onClick={() => {
+          console.log(control._fields);
+        }}
+      ></div>
+
+      <form
+        onSubmit={handleSubmit(onSubmit)}
+        style={{
+          height: '100vh',
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'space-around'
+        }}
+      >
```

Comment:
상준님 고생 많으셨습니다~!

제가 ImageUploder 제작할 때 useFormContext라는 API를 사용했어요. (바로 상위 부모 컴포넌트의 useForm return값을 props로 받지않고 사용할 수 있다고 합니다.)

TextField와 ImageUploder를 같이 쓰기 위해서
상위에서 useForm return 값 선언해주고 
FormProvider component로 감싸서 사용하면 동일하게 동작합니다!

```
  const methods = useForm();
  const { handleSubmit, control, register } = methods;
```

```suggestion
      <FormProvider {...methods}>
       <form
          onSubmit={handleSubmit(onSubmit)}
          style={{
            height: '100vh',
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'space-around'
          }}
        >
```

제가 FormProvider랑 form tag는 결합해서 쓸 수 있도록 custom component 한개 만들어서 추가로 pr 올릴게요!

[useFormContext 문서](https://react-hook-form.com/docs/useformcontext)

Reply:
오옹 useFormContext 사용하면 다른 컴포넌트들에 대해서도 Context 설정이 가능하군요!! 알겠습니다
다만 지금 validation 하는 방식이 TextField 내부적으로 진행하는 방식이라,  react hook form 내의 state까지는 반영되지 않을것 같습니다. 이부분 추가해서 커밋해서 올려보겠습니다~~